openssl/src/x509/mod.rs

```diff
@@ -425,6 +425,64 @@ impl X509Ref {
         }
     }
 
+    /// Returns this certificate's extensions for the given [Nid].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use openssl::nid::Nid;
+    /// use openssl::x509::X509;
+    ///
+    /// let cert = X509::from_pem(include_bytes!("../../test/extensions.pem")).unwrap();
+    /// let nid = Nid::create(&"1.3.6.1.4.1.41482.5.3", &"fw", &"firmware").unwrap();
+    /// let extension = cert.get_ext_by_nid(nid).unwrap().unwrap();
+    /// let value = extension.data().as_slice();
+    /// assert_eq!(value, [4, 3, 5, 2, 7]);
+    /// ```
+    #[corresponds(X509_get_ext_by_NID)]
+    pub fn get_ext_by_nid(&self, nid: Nid) -> Result<Option<&X509ExtensionRef>, ErrorStack> {
+        unsafe {
+            let loc = ffi::X509_get_ext_by_NID(self.as_ptr(), nid.as_raw(), -1);
+            Ok(if loc >= 0 {
+                Some(X509ExtensionRef::from_ptr(cvt_p(ffi::X509_get_ext(
```

Comment:
I'm kind of not sure if `cvt_p` is necessary here or whether returned null pointer can just be converted into None (that'd remove the Result from function signature, making it simpler). The docs doesn't seem to be conclusive on this matter: [X509_get_ext ](https://www.openssl.org/docs/man3.1/man3/X509_get_ext.html) :/